app/src/main/java/org/mozilla/fenix/components/toolbar/CFRPresenter.kt

```diff
@@ -0,0 +1,130 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.components.toolbar
+
+import android.content.Context
+import androidx.annotation.VisibleForTesting
+import androidx.compose.foundation.clickable
+import androidx.compose.material.Text
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.text.style.TextDecoration
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.cancel
+import kotlinx.coroutines.flow.filter
+import kotlinx.coroutines.flow.mapNotNull
+import kotlinx.coroutines.flow.transformWhile
+import mozilla.components.browser.state.selector.findCustomTabOrSelectedTab
+import mozilla.components.browser.state.store.BrowserStore
+import mozilla.components.browser.toolbar.BrowserToolbar
+import mozilla.components.lib.state.ext.flowScoped
+import mozilla.components.service.glean.private.NoExtras
+import org.mozilla.fenix.GleanMetrics.TrackingProtection
+import org.mozilla.fenix.R
+import org.mozilla.fenix.compose.cfr.CFRPopup
+import org.mozilla.fenix.compose.cfr.CFRPopupProperties
+import org.mozilla.fenix.ext.components
+import org.mozilla.fenix.settings.SupportUtils
+import org.mozilla.fenix.settings.SupportUtils.SumoTopic.TOTAL_COOKIE_PROTECTION
+import org.mozilla.fenix.theme.FirefoxTheme
+import org.mozilla.fenix.utils.Settings
+
+/**
+ * Delegate for handling all the business logic for showing CFRs in the toolbar.
+ *
+ * @param context [Context] used for various Android interactions.
+ * @param store [BrowserStore] which will be observed for tabs updates
+ * @param settings [Settings] allowing to read and write persistent user settings
+ * @param toolbar [BrowserToolbar] that will serve as anchor for the CFRs
+ * @param sessionId [String] optional custom tab id.
+ */
+class CFRPresenter(
+    private val context: Context,
+    private val store: BrowserStore,
+    private val settings: Settings,
+    private val toolbar: BrowserToolbar,
+    private val sessionId: String? = null
+) {
+    @VisibleForTesting
+    internal var tcpCfrScope: CoroutineScope? = null
+    @VisibleForTesting
+    internal var tcpCfr: CFRPopup? = null
```

Comment:
Keeping this as a nullable property ease testing with `assertNull` / `assertNotNull`.
The presenter is meant to be used for other CFRs also, naming this property just `popup` would almost assume it will have to be renamed later. Will add a `popup` suffix.

Reply:
Didn't think about testing, fair enough!